### PR TITLE
When the entrypoint receives SIGINT/SIGTERM, record an abort

### PR DESCRIPTION
**Pull request by smarterclayton**
Jenkins and Prow both track an "aborted" state, where either the job is terminated or a new job spec is available and the existing jobs should be stopped. Gubernator shows this state distinctly to make it easier to separate out user input (new commit pushed) from true flake failure.

At the current time, prow jobs using /entrypoint won't record aborted state if plank stops the pod (or if the kubelet evicts the pod, which is an abort as well).

Inside of entrypoint, treat SIGINT/SIGTERM as an abort and pass 130 (128+SIGINT) as the exit code, then in the sidecar treat exit code 130 as an abort request and write `"result":"ABORTED"` instead of `"result":"FAILURE"`.

@stevekuznetsov

### CODE REVIEW

**Comment by BenTheElder** [comment]
the lint failure looks to be https://github.com/bazelbuild/bazel/issues/5396 which has already been closed /cc @ixdy 

the build failure is legitimate and caused by the call to doUpload at line 61 in ` prow/sidecar/run.go`  not supplying the extra bool parameter.

File: prow/entrypoint/run.go [hunk 1]
```diff
@@ -36,6 +36,9 @@ const (
 	// InternalErrorCode is what we write to the marker file to
 	// indicate that we failed to start the wrapped command
 	InternalErrorCode = 127
+	// AbortedErrorCode is what we write to the marker file to
+	// indicate that we were terminated via a signal.
```
Comment (stevekuznetsov) [inline prow/entrypoint/run.go]: Shouldn't this be `128+signal`?

Comment (smarterclayton) [reply prow/entrypoint/run.go]: In this specific case we would need to behave the same for both signals.  So we either pass one exit code to mean abort or we check both.  Do we need to preserve the distinction between term and int?

Comment (stevekuznetsov) [reply prow/entrypoint/run.go]: Depends on how well-behaved we want to be.

Comment (smarterclayton) [reply prow/entrypoint/run.go]: If we think we’re the only reader of the marker file containing the status I’d say that the code is up to us.  We don’t actually signal that code back to the OS and kubelet

**Pull request merged**